src/os/ui/draw/drawcontrols.js

```diff
@@ -106,8 +178,80 @@ class Controller extends BaseDrawControlsCtrl {
   listenForMapReady() {
     googEvents.listenOnce(getIMapContainer(), MapEvent.MAP_READY, this.onMapReady, false, this);
   }
+
+  /**
+   * @inheritDoc
+   * @export
+   */
+  isActive() {
+    return super.isActive() || this.measuring;
+  }
+
+  /**
+   * Toggles the measure interaction.
+   * @param {boolean=} opt_value The toggle value
+   * @export
+   */
+  toggleMeasure(opt_value) {
+    var measure = this.getMeasureInteraction_();
+
+    if (measure) {
+      opt_value = opt_value !== undefined ? opt_value : !measure.getActive();
+
+      measure.setEnabled(opt_value);
+      measure.setActive(opt_value);
+      this.measuring = opt_value;
+    }
+  }
+
+  /**
+   * Handles draw end events.
+   * @param {DrawEvent} evt The draw event
+   * @private
+   */
+  onDrawEnd_(evt) {
+    if (evt.target instanceof Measure) {
+      this.toggleMeasure(false);
+    }
+  }
+
+  /**
+   * Gets the measure interaction from the map.
+   * @return {?Measure} The measure interaction
+   * @private
+   */
+  getMeasureInteraction_() {
+    var interactions = getMapContainer().getMap().getInteractions().getArray();
+    var measure = interactions.find((interaction) => {
+      return interaction instanceof Measure && interaction.isType('measure');
+    });
+    return /** @type {Measure} */ (measure);
+  }
+
+  /**
+   * Handles measure type changes.
+   * @param {MenuEvent<undefined>} evt The event from the menu
+   * @private
+   */
+  onMeasureTypeChange_(evt) {
+    Measure.method = /** @type {Method} */ (evt.type);
+    Settings.getInstance().set(this.key_, evt.type);
+    this.toggleMeasure(true);
+  }
 }
 
+/**
+ * Helper function for changiing the icons on measure options.
+ * @this {MenuItem}
+ */
+const updateIcons = function() {
+  if (this.eventType === Measure.method) {
```

Comment:
Do these still mean anything in context of the Draw menu? In the Measure menu they represented the "active" measure action we would use if you clicked the measure button, but we may want to use a measure type icon (ruler or drafting compass) now.